src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -372,7 +372,7 @@ protected function getUser()
     protected function isCsrfTokenValid(string $id, string $token): bool
     {
         if (!$this->container->has('security.csrf.token_manager')) {
-            throw new \LogicException('CSRF protection is not enabled in your application.');
+            throw new \LogicException('CSRF protection is not enabled in your application. Enable it with the "csrf_protection" key in "config/packages/framework.yaml"');
```

Comment:
Flex should be auto-enabling it when requiring the `security-csrf` package, no ? If yes, the right advice is to require it too

Reply:
That seems logical to me, bit I actually don't think it's enabled by default. It looks like it's always disabled unless you explicitly enable it: https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L107-L116

Reply:
😢 why making FrameworkBundle auto-enable all components except one ?

Reply:
Looks like something we need to fix.

Reply:
Anybody willing to submit a PR on that one? Would be cool to have it for 3.4/4.0 before final.

Reply:
@fapbot https://github.com/symfony/symfony/pull/25151